models/checks.go

```diff
@@ -21,31 +21,32 @@ var CheckHealthStatuses = []CheckHealthStatus{
 }
 
 type Check struct {
-	ID                 uuid.UUID           `json:"id" gorm:"default:generate_ulid()"`
-	CanaryID           uuid.UUID           `json:"canary_id"`
-	Spec               types.JSON          `json:"-"`
-	Type               string              `json:"type"`
-	Name               string              `json:"name"`
-	CanaryName         string              `json:"canary_name" gorm:"-"`
-	Namespace          string              `json:"namespace"  gorm:"-"`
-	Labels             types.JSONStringMap `json:"labels" gorm:"type:jsonstringmap"`
-	Description        string              `json:"description,omitempty"`
-	Status             CheckHealthStatus   `json:"status,omitempty"`
-	Uptime             Uptime              `json:"uptime"  gorm:"-"`
-	Latency            Latency             `json:"latency"  gorm:"-"`
-	Statuses           []CheckStatus       `json:"checkStatuses"  gorm:"-"`
-	Owner              string              `json:"owner,omitempty"`
-	Severity           string              `json:"severity,omitempty"`
-	Icon               string              `json:"icon,omitempty"`
-	Transformed        bool                `json:"transformed,omitempty"`
-	DisplayType        string              `json:"display_type,omitempty"  gorm:"-"`
-	LastRuntime        *time.Time          `json:"last_runtime,omitempty"`
-	NextRuntime        *time.Time          `json:"next_runtime,omitempty"`
-	LastTransitionTime *time.Time          `json:"last_transition_time,omitempty"`
-	CreatedAt          *time.Time          `json:"created_at,omitempty"`
-	UpdatedAt          *time.Time          `json:"updated_at,omitempty"`
-	DeletedAt          *time.Time          `json:"deleted_at,omitempty"`
-	SilencedAt         *time.Time          `json:"silenced_at,omitempty"`
+	ID                 uuid.UUID            `json:"id" gorm:"default:generate_ulid()"`
+	CanaryID           uuid.UUID            `json:"canary_id"`
+	Spec               types.JSON           `json:"-"`
+	Type               string               `json:"type"`
+	Name               string               `json:"name"`
+	CanaryName         string               `json:"canary_name" gorm:"-"`
+	Namespace          string               `json:"namespace"  gorm:"-"`
+	Labels             types.JSONStringMap  `json:"labels" gorm:"type:jsonstringmap"`
+	Description        string               `json:"description,omitempty"`
+	Status             CheckHealthStatus    `json:"status,omitempty"`
+	Uptime             Uptime               `json:"uptime"  gorm:"-"`
+	Latency            Latency              `json:"latency"  gorm:"-"`
+	Statuses           []CheckStatus        `json:"checkStatuses"  gorm:"-"`
+	Owner              string               `json:"owner,omitempty"`
+	Severity           string               `json:"severity,omitempty"`
+	Icon               string               `json:"icon,omitempty"`
+	TestThreshold      *types.TestThreshold `json:"testThreshold,omitempty" gorm:"column:test_threshold"`
```

Comment:
Shouldn't this be part of spec ?

Reply:
I think we should split the types into 2, 1 for the spec, and 1 for the results , The results wouldn't have any reference to threshold - it would only have a severity, and likewsie the spec wouldn't have status, uptime, latency, etc..

@yashmehrotra wdyt

Reply:
I'd prefer splitting due to separation of concerns

We keep `summary` in db, but never use that field, it always get calculated at runtime, should look into fixing that sometime as well